### ISSUE TITLE
Enhance star ratings with readonly/disabled states

### ISSUE THREAD

**Issue by julien-deramond** [Orange-OpenSource/Orange-Boosted-Bootstrap#1764]
Our Star rating component has a disabled version that can be found here: https://boosted.orange.com/docs/forms/checks-radios/#disabled-2.

The associated description begins with "Make star ratings look inactive inside or outside a form [...]".

IMO star rating should be enhanded to propose two separate states:
* `readonly` would have the same rendering as what we show in https://boosted.orange.com/docs/forms/checks-radios/#disabled-2
* `disabled` would show the orange stars as dark gray stars

A readonly state is different than a disabled state.

There's also in this spirit something to check with the readonly state because inputs in Bootstrap/Boosted are focusable in readonly state contrary to disabled state. In readonly, when it's focused, the value could be read by a11y tools as well.
